horrors.c

```diff
@@ -0,0 +1,86 @@
+// This file is a little story, you read it top to bottom left to right. It is a
+// journey of horror and revulsion. Whenever a new Terror is added to our little
+// bestiery, I _will_ be using it when convenient later. The point is to
+// illustrate the terror that is in the C language available to be misused.
+
+
+int const array_len = 4;
+// this is not allowed:
+// char buf[array_len];
+enum sizes {ZERO, ONE, TWO, THREE};
+// but this is:
+char but[TWO];
+// and confusingly this also is:
+char buu[ZERO];
+// or, equivalently (we are conveniently allowed to redeclare it)
+char buu[0];
+// you can _hold_ buu but never look inside
+// TODO
+
+
+
+int main() {
+  // BLOCKS
+
+  // Blocks are allowed inside functions, like so
+  int x = -1;
+  {
+    x += 1;
+    int x = 3;
+    int y = x + 1;
+  }
+  // naturally you can shadow variables, though the fact that you can shadow
+  // them and use the outer version both in the same block is a bit surprising
+
+  // STRUCTS
+  struct derp {int a, derp;};
+  // of course this is fine. Notice that you can reuse the name derp both as the
+  // struct tag and as a field in the struct.
+  struct derp derp;
+  derp.derp = 4;
```

Comment:
find some way to put a cast in there: `(struct derp) derp.derp`. Ideally make `derp` recursive having a field: `struct derp derp` instead of int

Reply:
```
struct derp {
	struct derp* derp;
};

struct derp derp;
derp.derp = &derp;

printf("here we go: %i\n", (struct derp*) ((struct derp)derp).derp);

```